packages/token-bridge-sdk/src/util/token.ts

```diff
@@ -202,3 +184,101 @@ export function getTokenLists(forL2ChainId?: string): TokenListWithId[] {
 
   return parsedStorage.filter(tokenList => tokenList.l2ChainId === forL2ChainId)
 }
+
+export function tokenListsToSearchableTokenStorage(
+  tokenLists: TokenListWithId[],
+  l1ChainId: string,
+  l2ChainId: string
+): SearchableTokenStorage {
+  return (
+    tokenLists
+      //
+      .reduce((acc: SearchableTokenStorage, tokenList: TokenListWithId) => {
+        tokenList.tokens.forEach(token => {
+          const address = token.address.toLowerCase()
+          const stringifiedChainId = String(token.chainId)
+
+          if (stringifiedChainId === l1ChainId) {
+            // The address is from an L1 token
+
+            if (typeof acc[address] === 'undefined') {
+              // First time encountering the token through its L1 address
+              acc[address] = {
+                ...token,
+                type: TokenType.ERC20,
+                l2Address: undefined,
+                tokenLists: []
+              }
+            } else {
+              // Token was already added to the map through its L2 token
+              acc[address] = { ...acc[address]!, address }
+            }
+
+            const tokenLists = acc[address]!.tokenLists
+
+            if (!tokenLists.includes(tokenList.bridgeTokenListId)) {
+              acc[address]!.tokenLists.push(tokenList.bridgeTokenListId)
+            }
+          } else if (stringifiedChainId === l2ChainId) {
+            // The token is an L2 token
+
+            if (!token.extensions?.bridgeInfo) {
+              return
+            }
+
+            // @ts-ignore
+            //
+            // TODO: should we upgrade '@uniswap/token-lists'?
```

Comment:
@OffchainLabs/full-stack should we do sth about this todo? seems like it's been here for a long time